experiments/modern/src/initialize.js

```diff
@@ -142,21 +275,28 @@ async function parseChildren(node, parsedParent, registry, zip) {
         return;
       }
       const parsedChild = await childParser(child, node, registry, zip);
-      if (child.children != null) {
-        await parseChildren(child, parsedChild, registry, zip);
+      const returnNode = { ...parsedChild, id: uuid() };
+      if (parsedChild.children != null) {
+        const parsedChildren = await parseChildren(parsedChild, registry, zip);
+        returnNode.children = parsedChildren.children;
       }
+      return returnNode;
     })
   );
-}
+  // remove comments other trimmed nodes
+  resolvedChildren = resolvedChildren.filter(item => item !== undefined);
```

Comment:
I personally prefer to use a new variable name in cases like this.

Reply:
yea, me too. I was starting to feel like the naming was getting silly like now I add `filteredResolvedChildren` or something, but probably better than mutating things :D